Automattic/woocommerce-payments
# Add Dispute Challenge and Accept buttons to Transaction Details

Fixes #6926 

> **Note**
> This PR is behind feature flag `_wcpay_feature_dispute_on_transaction_page`

#### Changes proposed in this Pull Request

* Adds dispute action buttons to the Transaction Details screen for disputes awaiting a response **(not inquiries)**, enabling the merchant to _Accept_ the dispute or begin/continue a _Challenge_ from this screen.
* Adds a test suite for the `DisputeDetails` component
* Introduces code quality improvements – updated and annotated some TS interfaces.

**Note**: this PR won't handle CTA buttons for inquiries – this will be handled in issue #7193.

**Screenshots**

Action buttons:

![dispute-actions](https://github.com/Automattic/woocommerce-payments/assets/3147296/74cb86af-3df6-4777-90b7-e3a30988c78d)

Accept modal:

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/bfe01917-4fb7-4653-9654-5e1f822a8031)

Snackbar on successful accept:
<img src="https://github.com/Automattic/woocommerce-payments/assets/3147296/7716d502-36da-482a-aeb9-8aa68f607a92" width="300px" />

Wide buttons on mobile (see design: RiEQmaKRI7u9USAI3lyZz0VX-fi-7430_30973):
<img src="https://github.com/Automattic/woocommerce-payments/assets/3147296/d04e9fb5-9644-4399-a870-941456efd285" width="300px" />

The below video demonstrates:
→ clicking the Challenge button
→ staging evidence via the challenge form
→ returning to the dispute details
→ opening the accept dispute modal
→ accepting the dispute

https://github.com/Automattic/woocommerce-payments/assets/3147296/89f5dc84-8c91-4c80-98b3-1f2604550e55


#### Testing instructions





* Create a disputed transaction using the card `4000000000000259` at checkout.
* Navigate to the disputed Transaction (Transactions → click the disputed transaction)
* Click **Challenge dispute** → You should be taken to the Dispute challenge screen.
* Click **Accept dispute** →  A modal will be shown. Confirm the text and styles are correct as per the design (RiEQmaKRI7u9USAI3lyZz0VX-fi-6974_70994).
	* Click **Cancel** to close the modal
	* Disable network / go offline in devtools (to force a network error)
	* Click **Accept dispute**
		* On error, a snackbar will show an error message in the bottom-left corner.
	* Re-enable network
	* Click **Accept dispute**
		* The modal should close
		* The dispute should be accepted
		* A snackbar will show a success message in the bottom-left corner
		* The Transaction Details screen will be hydrated with new data (`Disputed: Lost` status chip).
		* The dispute actions should no longer be available.
* View a dispute of any other status (won, lost, under review)
	* The dispute actions should not be visible
* View a disputed transaction for a dispute where a challenge has been staged (click "Save for Later" on the dispute challenge form.
	* The _Challenge_ button should be labelled `Continue with challenge`

-------------------


**Post merge**
